File: frontend/selenium-tests/pages/patientDashboardPage.js
const { By, until } = require('selenium-webdriver');
const config = require('../config');

class PatientDashboardPage {
    constructor(driver) {
        this.driver = driver;
        this.screenChildDashboard = By.id('screen-dashboard-child');
        this.welcomeMsg = By.id('child-welcome-msg');
        this.streakVal = By.id('child-streak');
        this.missionCard = By.xpath("//h3[contains(text(), \"Today's Mission\")]");
        this.pointsDisplay = By.id('points-display');
        this.rewardStoreBtn = By.xpath("//span[contains(text(), 'VIEW ALL')]");
        this.brushingModal = By.id('modal-brushing');
        this.btnStartBrush = By.id('btn-start-brush');
        this.timerVal = By.id('timer-val');
    }

    async isDashboardDisplayed() {
        const el = await this.driver.wait(until.elementLocated(this.screenChildDashboard), config.EXPLICIT_WAIT);
        return await el.isDisplayed();
    }

    async getWelcomeText() {
        const el = await this.driver.wait(until.elementLocated(this.welcomeMsg), config.EXPLICIT_WAIT);
        return await el.getText();
    }

    async openBrushingMission() {
        const card = await this.driver.findElement(this.missionCard);
        await card.click();
        await this.driver.wait(until.elementIsVisible(await this.driver.findElement(this.brushingModal)), config.EXPLICIT_WAIT);
    }

    async startBrushingTimer() {
        const btn = await this.driver.findElement(this.btnStartBrush);
        await btn.click();
    }
}

module.exports = PatientDashboardPage;

File: automation/pages/patient_dashboard_page.py
from selenium.webdriver.common.by import By
from automation.pages.base_page import BasePage

class PatientDashboardPage(BasePage):
    WELCOME_MSG = (By.ID, "child-welcome-msg")
    STREAK_TXT = (By.ID, "child-streak")
    START_MISSION_BTN = (By.ID, "btn-start-recording-main")
    SYNC_BTN = (By.XPATH, "//button[contains(text(), 'SYNC')]")
    LOGOUT_BTN = (By.XPATH, "//button[contains(text(), '⭐')]")
    POINTS_DISPLAY = (By.ID, "points-display")
    
    def is_dashboard_visible(self):
        return self.is_displayed(By.ID, "screen-dashboard-child")

    def get_welcome_text(self):
        return self.get_text(*self.WELCOME_MSG)

    def logout(self):
        self.click(*self.LOGOUT_BTN)
